firmware/components/driver_framebuffer/driver_framebuffer.c

```diff
@@ -283,33 +283,41 @@ uint32_t driver_framebuffer_getPixel(Window* window, int16_t x, int16_t y)
 	#endif
 }
 
-bool driver_framebuffer_flush(uint32_t flags)
+void driver_framebuffer_blit(Window* source, Window* target)
 {
-	if (!framebuffer) {
-		ESP_LOGE(TAG, "flush without alloc!");
-		return false;
+	if (source->vOffset >= source->height) return; //The vertical offset is larger than the height of the window
+	if (source->hOffset >= source->width)  return; //The horizontal offset is larger than the width of the window
+	for (uint16_t wy = source->vOffset; wy < source->drawHeight; wy++) {
+		for (uint16_t wx = source->hOffset; wx < source->drawWidth; wx++) {
+			if (wy >= source->height) continue; //Out-of-bounds

```

Comment:
`continue` tries the next pixel as well, but we already know that wy (or wx) are too large. I'd move the limits into the loop condition.